### ISSUE TITLE
Support `import { m } from '...'` on vscode extension

### ISSUE THREAD

**Issue by jeiea** [opral/inlang-paraglide-js#151]
### Background

I'm using paraglidejs with nextjs. To take advantage of the automatic import, I re-exported the contents of message.js, like `export * as m from './paraglide/messages'`.

### Expected behavior

Extension's translation preview should work.

### Actual behavior

Don't work.

### Additional context

I suspect [this line](https://github.com/opral/monorepo/blob/e74c771b0eb8c5aba68c3253a7078c2e3a1724fa/inlang/source-code/plugins/m-function-matcher/src/ideExtension/messageReferenceMatchers.ts#L22), can you do what the title asks?

**Comment by jeiea** [comment]
I could achieve this by modifying settings.json modules. I suggest changing the defaults.

**Comment by juliomuhlbauer** [comment]
Is there some problem using this approach, besides the extension.

**Comment by jeiea** [comment]
> Is there some problem using this approach, besides the extension.

Could you elaborate more? If 'this approach' refers to the `* as m` syntax, I don't prefer it because it requires additional work:

- I have to manually insert the import statement at the beginning of the file.

  - This can be made easier with snippet configuration, but it requires manual setup and learning.
  - File navigation with inserting is tedious.

**Comment by juliomuhlbauer** [comment]
I agree. I thought about the nextjs compiler/paraglide importing all messages. I don't think it is a problem right?

**Comment by jeiea** [comment]
> I don't think it is a problem right?

It's not really a problem but more of a feature request. I'm not suggesting to drop the `* as m` syntax, but to also support an alternative syntax.

**Comment by elron** [comment]
I agree, auto importing m isn't easy, I needed to manually go to the docs, search for the import line, copy and paste it into my code.

Auto import would be nice.

<img width="494" alt="Screen Shot 2024-08-12 at 4 43 16" src="https://github.com/user-attachments/assets/e0726179-72f5-451f-b6ae-cf20f2b6fe93">

<img width="589" alt="Screen Shot 2024-08-12 at 4 43 24" src="https://github.com/user-attachments/assets/d8505cb1-fe0f-461e-8fc4-6380f4d57a80">

**Comment by jeiea** [comment]
If there are no objections, is it okay to create a PR?

**Comment by juliomuhlbauer** [comment]
> If there are no objections, is it okay to create a PR?

Did you create the PR?

**Comment by jeiea** [comment]
Not yet.

**Comment by jeiea** [comment]
I'm currently using i18n-ally for vscode and have lost motivation to continue with this. I'm leaving it open for others.

**Comment by juliomuhlbauer** [comment]
> I'm currently using i18n-ally for vscode and have lost motivation to continue with this. I'm leaving it open for others.

But are you using Paraglide? I think is not supported by the extension

**Comment by samuelstroschein** [comment]
@jeiea if you do a PR, i'll quickly review and merge it

**Comment by ampcpmgp** [comment]
The following PR has been merged, but do I need to create other PR to reflect it in the sherlock extension?

https://github.com/opral/monorepo/pull/3367
https://github.com/opral/inlang-paraglide-js/issues/151

**Comment by samuelstroschein** [comment]
JSDelivr caches the latest version (which you probably use) for a week. Copy & paste the mfunction matcher plugin link you use in your project.settings `modules` into this purge [https://www.jsdelivr.com/tools/purge](https://www.jsdelivr.com/tools/purge) and then reload the inlang project / re-open sherlock

**Comment by samuelstroschein** [comment]
let me know if it works

**Comment by ampcpmgp** [comment]
Worked well! Thank you!

**Comment by samuelstroschein** [comment]
Paraglide JS 2.0 will ship with `import { m } from "./paraglide/message.js"` by default 🥳

**Comment by elron** [comment]
Wonderful!!
